File: ABC/B/260/B.rb
def judge_pass(max_num, subject, pass)
    cnt = 0
    subject.each do |score, i|
        break if cnt == max_num
        if !pass[i] then
            pass[i] = true
            cnt += 1 
        end
    end

    return pass
end

n, x, y, z = gets.chomp.split.map(&:to_i)
a = gets.chomp.split.map(&:to_i)
b = gets.chomp.split.map(&:to_i)
math = Array.new
eng = Array.new
sum = Array.new
pass = Hash.new(0)

n.times do |i| 
    math << [-a[i], i]
    eng << [-b[i], i]
    sum << [-a[i]-b[i], i]
    pass[i] = false
end

math.sort!
eng.sort!
sum.sort!

pass = judge_pass(x, math, pass)
pass = judge_pass(y, eng, pass)
pass = judge_pass(z, sum, pass)

pass.each do |key, val| p key+1 if val
end

File: ABC/B/260/B.py
def judge_pass(max_num, subject, pas):
    cnt = 0
    for score, i in subject:
        if cnt == max_num: break
        if not pas[i]:
            pas[i] = True
            cnt += 1

    return pas

def main():
    n, x, y, z = map(int, input().split())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    math = list()
    eng = list()
    sum = list()
    pas = dict()

    for i in range(n):
        math.append([-a[i], i])
        eng.append([-b[i], i])
        sum.append([-a[i]-b[i], i])
        pas[i] = False

    math = sorted(math)
    eng = sorted(eng)
    sum = sorted(sum)

    pas = judge_pass(x, math, pas)
    pas = judge_pass(y, eng, pas)
    pas = judge_pass(z, sum, pas)

    for key in pas:
        if pas[key]: print(key + 1)


if __name__ == '__main__':
    main()